SIPS/sip-25.md

```diff
@@ -0,0 +1,225 @@
+
+### Minimum Interface for Routing Snaps
+
+#### Registration with MetaMask
+
+- **Method**: `routingSnap_register`
+- **Description**: Registers the routing Snap with MetaMask for a specific CAIP-2 identifier(s), specifying the methods and events it can handle.
+- **Parameters**:
+  - `chains`: An array of CAIP-2 identifiers the Snap can handle.
+  - `methods`: An array JSON-RPC methods the Snap supports.
+  - `events`: An array of events the Snap can handle.
+
+#### Registering Account Snaps
+
+- **Method**: `accountSnap_register`
+- **Description**: Allows account Snaps to register with the routing Snap, specifying the CAIP-2(s) identifier and the methods and events they support.
+- **Parameters**:
+  - `caip2`:  An array of CAIP-2 identifiers the Snap can handle
+  - `methods`: An array of JSON-RPC methods the account Snap supports.
+  - `events`: An array of events the account Snap can handle.
+
+#### Handling Requests
+
+- **Method**: `handleCaip27Request`
+- **Description**: Interprets and routes CAIP-27 requests to an appropriate account Snap based on the CAIP-2 identifier and method.
+- **Parameters**:
+  - `scope`: The CAIP-2 identifier of the target chain.
+  - `request`: The JSON-RPC request object.
+- **Example Implementation**:
+  ```typescript
+  interface JsonRpcRequest {
+  method: string;
+  params: any[];
+  }
+
+  interface AccountSnap {
+    handleRequest(request: JsonRpcRequest): Promise<string>;
+  }
+
+  async function handleCaip27Request(scope: string, request: JsonRpcRequest): Promise<string> {
+    const { method, params } = request;
+    const { requiresSignature, account } = identifySignatureMethods(method, params);
+
+    if (requiresSignature) {
+      const accountSnap: AccountSnap | null = await findOrUserSelectAccountSnap(scope, method);
+      if (accountSnap) {
+        const signedTransaction: string = await accountSnap.handleRequest(request);
+        return await broadcastTransaction(signedTransaction);
+      } else {
+        throw new Error(`No account Snap found for scope: ${scope} and method: ${method}`);
+      }
+    } else {
+      return await sendDirectRequest(method, params);
+    }
+  }
+  ```
+- **Error Handling**: The method should include appropriate error handling and return clear error messages for scenarios such as unsupported methods, network issues, or signature failures.
+
+### Detailed Functionality
+
+1. **RPC routing Snap Registration**:
+   - RPC routing Snaps register with MetaMask via `routingSnap_register`,
+     specifying the CAIP-2 identifier(s) they can handle and the methods/events
+     they support.
+   - MetaMask maintains a registry of all RPC routing Snaps and their capabilities.
+
+2. **Account Snap Registration**:
+   - Account Snaps register with RPC routing Snaps using `accountSnap_register`, declaring their supported CAIP-2 identifiers, methods, and events.
+   - RPC routing Snaps maintain a registry of account Snaps and their capabilities.
```

Comment:
Why would this not fall on MetaMask to maintain this, e.g. via the PermissionController?